src/controls/LayerControl.js

```diff
@@ -120,37 +105,35 @@ export default class LayerControl extends Control {
     const container = document.createElement('div');
     container.classList.add('layer-selection', 'hidden');
     element.appendChild(container);
-    baseLayers.forEach(item => {
-      container.appendChild(item.thumbnailBox);
-      options.map.addLayer(item.layer);
-    });
 
     super({
       element: element,
       target: options.target,
     });
 
+    baseLayers.forEach(item => {
+      container.appendChild(item.thumbnailBox);
+    });
+
     this.layerControlElement = container;
     activatorButton.addEventListener(
       'click',
       this.toggleLayerSelection.bind(this),
       false,
     );
-    this.activatorButton = activatorButton;
-    this.map = options.map;
   }
 
-  /**
-   * @type {HTMLButtonElement} activator button, containing the layer selection icon
-   */
-  static activatorButton = null;
-
-  static map = null;
+  setMap(map) {
+    super.setMap(map);
+    this.getMap().addLayer(
+      new LayerGroup({ layers: baseLayers.map(i => i.layer) }),
```

Comment:
baseLayers should not be static, and if there is a previous map, the layer group should be removed. Otherwise layer duplication and reuse of stale renderers could render the map unusable.